docs/INTERNALS.md

```diff
@@ -0,0 +1,336 @@
+## Kotlin Compiler Plugin
+
+Anvil is a Kotlin compiler plugin. While this document describes the internals of Anvil,
+it assumes some prior knowledge of how to write a compiler plugin. Unfortunately, compiler
+plugins are still experimental and there aren’t many resources available. These resources with
+their sample projects, even though some are outdated, are helpful to get started:
+
+* [Fixing serialization of Kotlin objects once and for all](https://medium.com/bumble-tech/fixing-serialization-of-kotlin-objects-once-and-for-all-95886fddba7a)
+* [Writing Your Second Kotlin Compiler Plugin](https://bnorm.medium.com/writing-your-second-kotlin-compiler-plugin-part-1-project-setup-7b05c7d93f6c)
+* [Writing Your First Kotlin Compiler Plugin](https://twitter.com/KevinMost7/status/1050808104835829768?s=20&t=5yFTxnS-BnJdyiMqwlx2Nw)
+* [The Road to the New Kotlin Compiler](https://www.youtube.com/watch?v=iTdJJq_LyoY)
```

Comment:
Super Nit: If the link order here was changed to
* [The Road to the New Kotlin Compiler](https://www.youtube.com/watch?v=iTdJJq_LyoY)
* [Writing Your First Kotlin Compiler Plugin](https://twitter.com/KevinMost7/status/1050808104835829768?s=20&t=5yFTxnS-BnJdyiMqwlx2Nw)
* [Fixing serialization of Kotlin objects once and for all](https://medium.com/bumble-tech/fixing-serialization-of-kotlin-objects-once-and-for-all-95886fddba7a)
* [Writing Your Second Kotlin Compiler Plugin](https://bnorm.medium.com/writing-your-second-kotlin-compiler-plugin-part-1-project-setup-7b05c7d93f6c)

I think it would match up with a better natural viewing order for someone that's clicking through one-by-one in order. As someone without much existing knowledge on compiler plugins, I found those first two to be sufficient for reading the rest of the doc.

Reply:
I agree.